src/proxmoxsandbox/_impl/qemu_commands.py

```
@@ -363,6 +365,20 @@ async def configure_network_and_tags(
         vm_id: int,
         extra_tags: List[str] = [],
     ) -> None:
+        # Fetch VNets once and build complete mapping with zone info
+        vnet_details = {}
+        try:
+            all_vnets_data = await self.async_proxmox.request(
+                "GET", "/cluster/sdn/vnets"
+            )
+            vnet_details = {
+                vnet["alias"]: {"vnet": vnet["vnet"], "zone": vnet["zone"]}
+                for vnet in all_vnets_data
+                if "alias" in vnet and vnet["alias"]
```

This silently drops vnets that don't have alias or have it blank. 

Is that possible/desirable? It seems plausible that we should raise an error here rather than failing silently if there's some weird edge case.